tmt/steps/provision/podman.py

```diff
@@ -79,6 +79,13 @@ def wake(self) -> None:
         self.debug(
             f"Waking up container '{self.container}'.", level=2, shift=0)
 
+    def pull_image(self) -> None:
+        """ Pull image if not available or pull forced """
+        self.podman(
+            Command('pull', '-q', self.image),
```

Comment:
mypy is right here: `self.image` is indeed `Optional[str]`, which means it may be set to `None`, while `Command` accepts only `str` or `Path` arguments. `None` is neither of these two.

Note line 87 in the original file, `assert self.image is not None` - this tells mypy to narrow the type: if we as developers are sure the `self.image` is not `None`, using `assert` hints mypy about this fact. Then it's clear `self.image` is a plain `str` and therefore acceptable as a `Command` parameter.

Please, add `assert self.image is not None  # narrow type` into the `pull_image` method, that should appease mypy.